### PR TITLE
Simplify imagelist

### PR DESCRIPTION
**Pull request by vadz**
The main change here is getting rid of wxOSX-specific `wxImageList` implementation completely as it was almost (but not quite, as usual, for maximal fun) identical to the generic one. So I've merged them together and then modernized and simplified the generic version a lot, notably getting rid of all the casts due to using type unsafe `wxObjectList`.

The only change I have some doubts about is the part of the last one in which I switched from storing `wxBitmap*` to just `wxBitmap` objects, i.e. never storing `wxIcon*` any more, but just converting to them if necessary. AFAICS this shouldn't be a problem because such conversions should be lossless and efficient (i.e. the new `wxIcon` will just reference the same `wxBitmapRefData` as the existing `wxBitmap`) and `wxDC::DrawIcon()` does the same thing as `DrawBitmap()` anyhow. But please let me know if I'm missing anything.

TIA!

### CODE REVIEW

File: include/wx/generic/imaglist.h [hunk 1]
```diff
@@ -10,48 +10,57 @@
...
 
     virtual bool Draw(int index, wxDC& dc, int x, int y,
               int flags = wxIMAGELIST_DRAW_NORMAL,
               bool solidBackground = false);
 
-    // Internal use only
-    const wxBitmap *GetBitmapPtr(int index) const;
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED_MSG("Don't use this unportable and doing nothing overload")
+    bool Create() { return true; }
```
Comment (paulcor) [inline include/wx/generic/imaglist.h]: Seems like this should be OSX-only, as it doesn't currently exist anywhere else. Also, the message is rather awkward English, I would suggest something like "Don't use this overload, it's not portable and does nothing".

**Review by paulcor — APPROVED** [review]

File: include/wx/generic/imaglist.h (hunk 1 above)
Comment (vadz) [reply include/wx/generic/imaglist.h]: It does exist in the [generic version](https://github.com/wxWidgets/wxWidgets/blob/v3.0.0/include/wx/generic/imaglist.h#L28).

I'll fix the deprecation message, thanks.

**Comment by vadz** [comment]
@csomor If you don't have any objections, I'm going to merge this soon.

**Comment by csomor** [comment]
I'm fine with it, I still think I might have to add some scaling to wxIcon handling, but simplifying and having one code base is the way to go IMHO, so for me it's a go ahead!

**Pull request merged**